include/unifex/type_traits.hpp

```diff
@@ -51,18 +82,36 @@ using instance_of = std::bool_constant<instance_of_v<T, X>>;
 
 struct unit {};
 
+template <bool B>
+struct _conditional {
+  template <typename, typename T>
+  using apply = T;
+};
+template <>
+struct _conditional<true> {
+  template <typename T, typename>
+  using apply = T;
+};
+
+template <bool B, typename T, typename U>
+using conditional_t = typename _conditional<B>::template apply<T, U>;
+
 template <typename T>
-using non_void_t = std::conditional_t<std::is_void_v<T>, unit, T>;
+using non_void_t = conditional_t<std::is_void_v<T>, unit, T>;
 
 template <typename T>
-using wrap_reference_t = std::conditional_t<
+using wrap_reference_t = conditional_t<
     std::is_reference_v<T>,
     std::reference_wrapper<std::remove_reference_t<T>>,
     T>;
 
+template <typename Class, typename Member>
+using member_t =
+    conditional_t<std::is_lvalue_reference_v<Class>, const Member&, Member>;
```

Comment:
Could we avoid instantiating `conditional_t` and `is_lvalue_reference_v` here through something like the following?
```
template<typename Class, typename Self, typename Member>
using member_t = decltype(((((Self(*)())nullptr)()).*((Member Class::*(*)())nullptr)()) );
```

And usage like: `member_t<type, Self, Value>`